lib/api/model/model.dart

```diff
@@ -493,26 +500,72 @@ sealed class Message {
   @JsonKey(name: 'match_subject')
   final String? matchTopic;
 
-  static MessageEditState _messageEditStateFromJson(dynamic json) {
+  static MessageEditState _messageEditStateFromJson(Object? json) {
     // This is a no-op so that [MessageEditState._readFromMessage]
     // can return the enum value directly.
     return json as MessageEditState;
   }
 
-  static Reactions? _reactionsFromJson(dynamic json) {
-    final list = (json as List<dynamic>);
+  static Reactions? _reactionsFromJson(Object? json) {
+    final list = (json as List<Object?>);
     return list.isNotEmpty ? Reactions.fromJson(list) : null;
   }
 
   static Object _reactionsToJson(Reactions? value) {
     return value ?? [];
   }
 
-  static List<MessageFlag> _flagsFromJson(dynamic json) {
-    final list = json as List<dynamic>;
+  static List<MessageFlag> _flagsFromJson(Object? json) {
+    final list = json as List<Object?>;
     return list.map((raw) => MessageFlag.fromRawString(raw as String)).toList();
   }
 
+  static List<Submessage>? _submessagesFromJson(Object? json) {
+    final list = json as List<Object?>;
+    return list.isNotEmpty
+      ? list.map((e) => Submessage.fromJson(e as Map<String, Object?>)).toList()
+      : null;
+  }
+
+  static Poll? _readPoll(Map<Object?, Object?> json, String key) {
+    final submessages = _submessagesFromJson(json[key]);
+    // If empty, [submessages] is converted to null.
+    assert(submessages == null || submessages.isNotEmpty);
+    if (submessages == null) return null;
+
+    final messageId = (json['id'] as num).toInt();
+    assert(submessages.every((e) => e.messageId == messageId));
+
+    final senderId = (json['sender_id'] as num).toInt();
+    assert(submessages.first.senderId == senderId);
+
+    final widgetData = WidgetData.fromJson(submessages.first.content);
+    try {
+      switch (widgetData) {
+        case PollWidgetData():
+          return Poll.fromSubmessages(
+            submessages,
+            senderId: senderId,
+            widgetData: widgetData
+          );
+        case UnsupportedWidgetData():
+          assert(debugLog('Unsupported widgetData: ${widgetData.json}'));
+      }
+    } on TypeError catch (e) {
+      assert(debugLog('Malformed data of widget type ${widgetData.widgetType}: $e\n${jsonEncode(submessages)}')); // TODO(log)
+    }
```

Comment:
Why swallow this error? Seems like that'll lead to us having a `Message` object that we're unaware has a special meaning.